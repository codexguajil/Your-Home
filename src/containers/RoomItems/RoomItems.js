import React from 'react';
import { Collapse } from 'antd';
import NewMaterial from '../NewMaterial/NewMaterial';
import EditMaterial from '../EditMaterial/EditMaterial';

const Panel = Collapse.Panel;

function callback(key) {
  console.log(key);
}

class RoomItems extends React.Component {
  state = { expanded: false };

  renderMaterialTypes = () => {
    let allCategories = []
    this.props.materials.forEach(material => {
      if (!allCategories.includes(material.element_type)) {
        allCategories.push(material.element_type)
      }
    })

    return allCategories.map((category, i) => {
      return(
        <Panel header={category} key={i}>
          {this.renderMaterials(category)}
        </Panel>
      )
    })
  }

  renderMaterials = (type) => {
    let materials = this.props.materials.filter(material => (material.element_type === type))
    return materials.map((material, i) => {
      const { brand, manual_url, model_number, name, notes, quantity, unit_price, vendor } = material.material
        return(
          <div key={i} className='material'>
            <p>{name}, {brand}</p>
            <div className='material-control'>
              <EditMaterial />
              <button className='delete-material-btn'><i className="fas fa-trash-alt"></i></button>
            </div>
          </div>
        )
    })
  }

  render() {
    return (
      <div>
        <div>
          <Collapse onChange={callback}>
            {this.renderMaterialTypes()}
          </Collapse>
        </div>
        <div className='add-material-btn'>
          <NewMaterial />
        </div>
      </div>
    );
  }
}

export default RoomItems;