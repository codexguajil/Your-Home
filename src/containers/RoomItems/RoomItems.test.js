import React from 'react';
import { shallow } from 'enzyme';
import RoomItems from './RoomItems';

describe('RoomItems', () => {
  let wrapper;
  let mockMaterials;
  
  beforeEach(() => {
    mockMaterials = [
      {
          "id": 1,
          "element_type": "Flooring",
          "material": {
              "id": 1,
              "name": "Material 1",
              "model_number": "abc1",
              "brand": "Kenmoore",
              "vendor": "HD",
              "manual_url": null,
              "notes": null,
              "quantity": null,
              "unit_price": null
          }
      },
      {
          "id": 2,
          "element_type": "Flooring",
          "material": {
              "id": 2,
              "name": "Material 2",
              "model_number": "abc2",
              "brand": "Kenmoore",
              "vendor": "HD",
              "manual_url": null,
              "notes": null,
              "quantity": null,
              "unit_price": null
          }
      },
      {
        "id": 3,
        "element_type": "Fixtures",
        "material": {
            "id": 3,
            "name": "Material A",
            "model_number": "abc1",
            "brand": "GE",
            "vendor": "HD",
            "manual_url": null,
            "notes": null,
            "quantity": null,
            "unit_price": null
        }
    }
  ]
    wrapper = shallow(<RoomItems materials={mockMaterials} />)
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should collect all categories when renderMaterialTypes is called', () => {
    let result = wrapper.instance().renderMaterialTypes()
    expect(result.length).toBe(2)
  });

  it('should render category panels for each category collected when renderMaterialTypes is called', () => {

  });

  it('should render materials when renderMaterials is called', () => {

  });
});