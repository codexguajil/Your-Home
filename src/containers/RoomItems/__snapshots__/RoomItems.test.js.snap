// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoomItems should match snapshot 1`] = `
<Collapse
  bordered={true}
  expandIconPosition="left"
  onChange={[Function]}
  openAnimation={
    Object {
      "appear": [Function],
      "enter": [Function],
      "leave": [Function],
    }
  }
>
  <CollapsePanel
    header="Flooring"
    key="0"
  >
    <p
      key="0"
    >
      Material 1
      , 
      Kenmoore
       
      <button>
        edit
      </button>
    </p>
    <p
      key="1"
    >
      Material 2
      , 
      Kenmoore
       
      <button>
        edit
      </button>
    </p>
  </CollapsePanel>
  <CollapsePanel
    header="Fixtures"
    key="1"
  >
    <p
      key="0"
    >
      Material A
      , 
      GE
       
      <button>
        edit
      </button>
    </p>
  </CollapsePanel>
</Collapse>
`;
